test/regressions/index.js

```diff
@@ -225,21 +225,31 @@ if (unusedBlacklistPatterns.size > 0) {
 const viewerRoot = document.getElementById('test-viewer');
 
 function FixtureRenderer({ component: FixtureComponent }) {
+  const viewerReactRoot = React.useRef(null);
+
   React.useLayoutEffect(() => {
-    const children = (
-      <TestViewer>
-        <FixtureComponent />
-      </TestViewer>
-    );
+    const renderTimeout = setTimeout(() => {
+      const children = (
+        <TestViewer>
+          <FixtureComponent />
+        </TestViewer>
+      );
 
-    ReactDOM.render(children, viewerRoot);
-  }, [FixtureComponent]);
+      if (viewerReactRoot.current === null) {
+        viewerReactRoot.current = ReactDOMClient.createRoot(viewerRoot);
+      }
+
+      viewerReactRoot.current.render(children);
+    });
 
-  React.useLayoutEffect(() => {
     return () => {
-      ReactDOM.unmountComponentAtNode(viewerRoot);
+      clearTimeout(renderTimeout);
+      setTimeout(() => {
```

Comment:
React tree can't be synchronously unmounted during render, so I introduced setTimeout to run this code after React lifecycle completes. Taken from https://stackoverflow.com/questions/73459382/react-18-async-way-to-unmount-root#answer-74445760